### PR TITLE
Implement auto-close creatives on nav away

### PR DESCRIPTION
**Pull request by rsmith-attentive**
- Auto closes creative on nav away
- Updates example app to have tab nav for testing the above behavior
- Adds scroll to example app
- adds test for new behavior
- More logging to help debugging efforts

### CODE REVIEW

File: Sources/ATTNSDK.m [hunk 1]
```diff
@@ -100,9 +101,10 @@ - (void)trigger:(UIView *)theView handler:(ATTNCreativeTriggerCompletionHandler)
 
   [[wkWebViewConfiguration userContentController] addScriptMessageHandler:self name:@"log"];
 
-  NSString *userScriptWithEventListener = @"window.addEventListener('message', (event) => {if (event.data && event.data.__attentive && event.data.__attentive.action === 'CLOSE') {window.webkit.messageHandlers.log.postMessage(event.data.__attentive.action);}}, false);";
+  NSString *userScriptWithEventListener = @"window.addEventListener('message', (event) => {if (event.data && event.data.__attentive) {window.webkit.messageHandlers.log.postMessage(event.data.__attentive.action);}}, false);window.addEventListener('visibilitychange', (event) => {window.webkit.messageHandlers.log.postMessage(`Document Hidden: ${document.hidden}`);}, false);";
```
Comment (TWaltze) [inline Sources/ATTNSDK.m:104]: I suggest moving the message `Document Hidden: ${document.hidden}` to a variable/function that we can template. The message as it stands is a little too "rich" in my opinion (various capitalizations, punctuation, spacing, etc) to be manually matched against down below. Maybe generate the string using a function like `generateVisibilityEvent(visibility: boolean): string`

File: ExampleSwift/Podfile [hunk 2]
```diff
@@ -5,6 +5,6 @@ target 'ExampleSwift - Pod' do
   use_frameworks!
 
   # Pods for Example
-  pod 'attentive-ios-sdk'
+  pod 'attentive-ios-sdk', '0.4.3-beta.1'
```
Comment (TWaltze) [inline ExampleSwift/Podfile:8]: Are we always going to want to define the exact version for the example app now? So in future version changes we'll want to bump this as well?

File: Sources/ATTNSDK.m (hunk 1 above)
Comment (rsmith-attentive) [reply Sources/ATTNSDK.m:104]: good point, will update

File: ExampleSwift/Podfile (hunk 2 above)
Comment (rsmith-attentive) [reply ExampleSwift/Podfile:8]: I don't have a strong conviction either way. This change was to test out the beta functionality of cocoapods, but I can revert now that we know it works as expected.

**Review by TWaltze — DISMISSED** [review]

**Review by TWaltze — APPROVED** [review]

**Pull request merged**